mmf/trainers/base_trainer.py

```diff
@@ -242,22 +242,24 @@ def train(self):
             if self.current_epoch > self.max_epochs:
                 break
 
-            for batch in self.train_loader:
+            epoch_num_batches = len(self.train_loader)
+
+            for epoch_batch_index, batch in enumerate(self.train_loader):
                 self.profile("Batch load time")
-                self.current_iteration += 1
-                self.writer.write(self.num_updates + 1, "debug")
+
+                self._check_start_update(epoch_batch_index, epoch_num_batches)
 
                 report = self._forward_pass(batch)
                 loss = self._extract_loss(report)
                 self._backward(loss)
-                should_break = self._logistics(report)
 
-                if self.num_updates > self.max_updates:
-                    should_break = True
+                should_break = self._check_finish_update(epoch_batch_index, report)
 
                 if should_break:
                     break
 
+                epoch_batch_index += 1
```

Comment:
Does this need to be incremented here? Wouldn't enumerate automatically increment this?